### PR TITLE
BasP-MR-14: methods GetAll, GetbyId and Post

### PR DESCRIPTION
**Pull request by VictoriaNegrotto**
I worked in validations, model, routes and controllers of Classes with methods GetAll, GetbyId and Post

### CODE REVIEW

File: src/controllers/classes.js [hunk 1]
```diff
@@ -1,94 +1,69 @@
...
-  });
-});
+const getAllClasses = (req, res) => {
+  Class.find()
+    .then((classes) => res.status(200).json({
+      message: 'Class list',
+      data: classes,
+      error: false,
+    }))
+    .catch((error) => res.status(400).json({
```
Comment (LucianoStradiot) [inline src/controllers/classes.js:12]: This status should be an error 500 because it comes from the server

File: src/models/Class.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+const ClassSchema = new Schema({
+  day: {
+    type: String,
+    enum: ['Monday', 'Wenesday', 'Friday'],
+  },
+  hour: {
+    type: Number,
+    timestamps: true,
+  },
+  trainerId: {
```
Comment (LucianoStradiot) [inline src/models/Class.js:14]: In this resources should be a require property for extra validation

File: src/validations/classes.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const Joi = require('joi');
+
+const validateCreation = (res, req, next) => {
```
Comment (LucianoStradiot) [inline src/validations/classes.js:3]: This const could be called validateClassCreation to be consistent with the code of the other resources of the app

File: src/validations/classes.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    trainerId: Joi.string().min(4).max(15).required(),
+    activityId: Joi.string().min(4).max(15).required(),
+    slots: Joi.number().required(),
+  });
+
+  const validation = classValidation.validate(req.body);
+
+  if (!validation.error) return next();
+  return res.status(400).json({
+    messege: `Error: ${validation.error.details[0].msg}`,
```
Comment (Eugeniagassman) [inline src/validations/classes.js:16]: is message not messege. And in the message when you make the array it should not be msg, but message to be consistent.

**Review by LucianoStradiot — APPROVED** [review]
Looks good!

File: src/models/Class.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    type: String,
+  },
+  activityId: {
+    type: String,
+  },
+  slots: {
+    type: Number,
+  },
```
[the remark this answers is not in the document]
Comment (VictoriaNegrotto) [reply src/models/Class.js]: yes, i put in the last commit. thanks!!!

File: src/controllers/classes.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    .catch((error) => res.status(500).json({
+      message: 'Error',
+      error,
+    }));
+};
+
+const getClassById = (req, res) => {
+  const { id } = req.params;
+  if (!id.match(regexObjectId)) {
+    res.status(404).json({
```
Comment (Juanoauil) [inline src/controllers/classes.js:21]: it should return this response because we do not want execute endpoint

File: src/controllers/classes.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+      message: 'ID invalid, please correct',
+      data: undefined,
+      error: true,
+    });
+  }
+  Class.findById(id)
+    .then((classes) => {
+      if (!classes) {
+        res.status(404).json({
```
Comment (Juanoauil) [inline src/controllers/classes.js:30]: Same here

**Review by Juanoauil — APPROVED** [review]

**Pull request merged**